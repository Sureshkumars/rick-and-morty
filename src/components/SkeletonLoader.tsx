import React from "react";
import { Box, SkeletonCircle, SkeletonText } from "@chakra-ui/core";

export const SkeletonLoader: React.FC = () => {
  return (
    <Box padding="6" boxShadow="lg" bg="white" minWidth="300px">
      <SkeletonCircle size="10" />
      <SkeletonText mt="4" noOfLines={4} spacing="4" />
    </Box>
  );
};
