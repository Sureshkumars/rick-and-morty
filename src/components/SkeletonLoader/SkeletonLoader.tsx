import React from "react";
import { Box, SkeletonCircle, SkeletonText } from "@chakra-ui/core";

export const SkeletonLoader: React.FC = () => {
  return (
    <Box
      padding="6"
      boxShadow="lg"
      bg="white"
      minWidth="300px"
      aria-label="skeleton-container"
    >
      <SkeletonCircle size="10" />
      <SkeletonText mt="4" noOfLines={4} spacing="4" />
    </Box>
  );
};
