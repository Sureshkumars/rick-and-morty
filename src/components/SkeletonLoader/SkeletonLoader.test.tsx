import React from "react";
import { SkeletonLoader } from "./SkeletonLoader";
import { render } from "@testing-library/react";

it("Loader Skeleton matches snapshot", () => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
  const { asFragment }: any = render(<SkeletonLoader />);
  expect(asFragment()).toMatchSnapshot();
});
