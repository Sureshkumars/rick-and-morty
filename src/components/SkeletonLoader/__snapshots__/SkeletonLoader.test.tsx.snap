// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Skeleton matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="skeleton-container"
    class="css-m4budz"
  >
    <div
      class="chakra-skeleton css-1y9avnh"
    />
    <div
      class="chakra-skeleton__group css-1fhgjcy"
    >
      <div
        class="chakra-skeleton css-1pzxi2p"
      />
      <div
        class="chakra-skeleton css-1pzxi2p"
      />
      <div
        class="chakra-skeleton css-16x1prk"
      />
    </div>
  </div>
</DocumentFragment>
`;
